packages/utils/telemetry-utils/src/batchedTelemetryHelper.ts

```diff
@@ -0,0 +1,73 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import type {
+	ITelemetryGenericEventExt,
+	ITelemetryLoggerExt,
+	ITelemetryPerformanceEventExt,
+} from "./telemetryTypes.js";
+
+/**
+ * Expected type of the custom data passed into the logger.
+ */
+type ICustomDataMap = Record<string, number>;
+
+/**
+ * Helper class that accumulates custom data and sends it to the logger every time a specified number (i.e., `threshold`) of logs is reached.
+ */
+export class BatchedTelemetryHelper {
```

Comment:
Should this be a logger implementation? Seems like this should be usable as an `ITelemetryLogger`, which just has the property that it accumulates and pushes based on a threshold.

Reply:
I considered that when I was chatting with Ji and can't recall now why exactly I shied away from it... One question to figure out in that case is: do all numeric properties in the event that gets passed to the logging method get averaged out every time? Do you have to provide a new parameter to specify which ones (in which case that method signature would not match the normal telemetry logger interface)?

Reply:
Wouldn't it be strange for an instance of this type to be used as a normal logger? Seems odd to have an object fulfill two different contracts.

Reply:
> Wouldn't it be strange for an instance of this type to be used as a normal logger? Seems odd to have an object fulfill two different contracts.

I wasn't suggesting we force this class to implement the logger interface. I was suggesting that we could implement the desired behavior as a logger so that it could be reused by existing infrastructure. Just like we have a ConsoleLogger that outputs logs to the console, we could have a BatchedLogger that performs the batching logic we want.